### PR TITLE
Mining Equipment Locker is now a vending machine

### PR DESCRIPTION
**Pull request by ghost**
The Mining Equipment Locker was a pretty huge snowflake. It was basically a vending machine in every single aspect, except the part where it was supposed to be coded as one

Now, it's just a vending machine, period. No reason to do otherwise
- Added Dwarven Mining Equipment, the vending machine that replaces Mining equipment lockers. It requires Mining access and vends everything the old machine used to vend, except for duckies and cigars
- The vending machine is linked to the Cargo account to emulate the old vending machine's behavior. Not that it will function correctly, given that the Cargo account is depreciated. This should only affect where the money is going (otherwise it's supposed to go to the station's account)
- The prices have been set way down. You can most likely deck out decently with your starting money. Some prize items like jetpacks and hookshots cost way more than your starting money, but then that's what Mining is for
- Expansion of items in the vending machine is possible, also maybe moving the machines on the outpost altogether
- All items from the old Mining Equipment Locker file was now moved to mine_items.dm. The old mining_equipment.dm itself now only contains the Ore Redemption Machine, and was renamed to this effect
- A recharge pack was added to use with the machine
- Maps updated to use the new vending machine, and Mining equipment's sprites were recycled whenever applicable

### CODE REVIEW

File: code/game/machinery/vending.dm [hunk 1]
```diff
@@ -2128,3 +2128,58 @@
...
+	req_access_txt = "48"
+	products = list(
+		/obj/item/weapon/reagent_containers/food/snacks/hotchili = 10,
+		/obj/item/weapon/reagent_containers/food/drinks/bottle/whiskey = 10,
+		/obj/item/device/wormhole_jaunter = 10,
+		/obj/item/weapon/gun/energy/kinetic_accelerator = 10,
+		/obj/item/weapon/pickaxe/jackhammer = 5,
+		/obj/item/weapon/resonator = 5,
+		/mob/living/simple_animal/hostile/mining_drone/ = 5,
+		/obj/item/weapon/tank/jetpack/carbondioxide/ = 3,
```
Comment (N3X15) [inline code/game/machinery/vending.dm]: Remove the slash at the end of the type

File: code/game/machinery/vending.dm [hunk 2]
```diff
@@ -2128,3 +2128,58 @@
...
+		)
+	prices = list(
+		/obj/item/weapon/reagent_containers/food/snacks/hotchili = 10,
+		/obj/item/weapon/reagent_containers/food/drinks/bottle/whiskey = 25,
+		/obj/item/device/wormhole_jaunter = 25,
+		/obj/item/weapon/gun/energy/kinetic_accelerator = 150,
+		/obj/item/weapon/pickaxe/jackhammer = 250,
+		/obj/item/weapon/resonator = 50,
+		/mob/living/simple_animal/hostile/mining_drone/ = 100,
+		/obj/item/weapon/tank/jetpack/carbondioxide/ = 500,
```
Comment (N3X15) [inline code/game/machinery/vending.dm]: here, too

**Comment by N3X15** [comment]
Changelog?

**Comment by Kurfursten** [comment]
>The Cargo account is depreciated
Since when?

Just this week I used it. In fact, it's a popularity known use of the Quartermaster to buy stuff from the locker at shift start using Cargo Account money, then withdraw that money again from the account. Free particle accelerators and hookshots.

**Comment by dylanstrategie** [comment]
Will push a changelog soon

**Comment by Shadowmech88** [comment]
Am I to understand that hookshots will now be 1000 credits?
That's even cheaper than they were at first, and the price was raised in the first place to prevent miners from buying a shitton and spreading them all over the station.

I also don't like that it requires mining access. Now instead of a department head being able to ask the AI to let them in and buy a lazarus injector or whatever they want using their department funds, they now have to track down someone with enough access to give their ID mining access.

I also disagree with removal of the, as far as I know, only renewable source of soap on the station.

File: code/game/machinery/vending.dm [hunk 3]
```diff
@@ -2128,3 +2128,58 @@
...
 	pack = /obj/structure/vendomatpack/nuka
+
+/obj/machinery/vending/mining
+	name = "Dwarven Mining Equipment"
+	desc = "Get your mining equipment here, and above all keep digging!"
+	product_slogans = "This asteroid isn't going to dig itself!;Stay safe in the tunnels, bring two Kinetic Accelerators!;Jetpacks, anyone?"
+	product_ads = "Hungry, thirsty or unequipped? We have your fix!"
+	vend_reply = "What a glorious time to mine!"
+	icon_state = "mining"
+	req_access_txt = "48"
```
Comment (PJB3005) [inline code/game/machinery/vending.dm]: oh use fucking defines you oaf.

**Comment by PJB3005** [comment]
:-1: removing the soap.

**Comment by dylanstrategie** [comment]
It would be a better idea to add a general vendor that can dispense items like soap, when I think of places where I can get soap bars, Mining Vendors certainly isn't the first thing that comes to mind

**Comment by PJB3005** [comment]
@dylanstrategie then remove the soap WHEN you make that vendor, not now you oaf.

**Comment by Probe1** [comment]
We literally shot this down on gitlab and you remade it on github.
https://gitlab.com/vgstation/vgstation/merge_requests/195

This is the exact same branch as the previous version without anything changed isn't it?  What makes you think everyones extremely negative reaction changed?

**Comment by dylanstrategie** [comment]
Because you are 100 % incorrect

Read the code before commenting, your complaint was that I stripped Cargo's money cow. Now the machine is linked to cargo's account. I made many changes that I could not push to the GitLab version due to the MR issue

**Comment by PJB3005** [comment]
@dylanstrategie more like 75%

File: code/game/machinery/vending.dm (hunk 3 above)
Comment (dylanstrategie) [inline code/game/machinery/vending.dm]: Defines for what, the access ? Are there access defines ?

**Comment by dylanstrategie** [comment]
Cite me an re-occurring issue that isn't the soap

**Comment by Probe1** [comment]
Dylan no one wanted ANY of your changes.  The cargo department account just showed that you have NO FUCKING IDEA what you're talking about since you thought it was broken (it isn't) and that it isn't used (it is). 

>Added Dwarven Mining Equipment, the vending machine that replaces Mining equipment lockers. It requires Mining access and vends everything the old machine used to vend, except for soap, duckies and cigars

Removing content for no reason

>The vending machine is linked to the Cargo account to emulate the old vending machine's behavior. Not that it will function correctly, given that the Cargo account is depreciated. This should only affect where the money is going (otherwise it's supposed to go to the station's account)

Not broken

>The prices have been set way down. You can most likely deck out decently with your starting money. Some prize items like jetpacks and hookshots cost way more than your starting money, but then that's what Mining is for

We just spent 3 weeks arguing with deity over the prices being too little for the hookshot and you want to reduce it now!

>Expansion of items in the vending machine is possible, also maybe moving the machines on the outpost altogether

already existed, fluff

>All items from the old Mining Equipment Locker file was now moved to mine_items.dm. The old mining_equipment.dm itself now only contains the Ore Redemption Machine, and was renamed to this effect

N/A

>A recharge pack was added to use with the machine
Maps updated to use the new vending machine, and Mining equipment's sprites were recycled whenever applicable

It works perfectly fine as it is and miners sure as fuck not interested in refilling the vending machine so they can get more laz injectors.


Stop making balance PRs without paying any attention to what people want.

File: code/game/machinery/vending.dm (hunk 3 above)
Comment (PJB3005) [inline code/game/machinery/vending.dm]: access.dm, use the vars there (yes they're not defines but they act as them, effectively)

**Comment by dylanstrategie** [comment]
Probe, this isn't a balance PR. I honestly don't see what kind of mental gymnastics you went through to see a balance PR. This is a refactor PR, aka turning the mining equipment machine into a vending machine

The only things that functionally changed are the following : 

- You need Mining access to use it, or you can snap wires until you drop the ID requirement
- The duckie and cigar were removed. If you really want duckies, that can't possibly be something the gods won't give you on a half-decent prayer, and cigars can be obtained in public vendors for a coin
- The prices are much lower. Since this would be the first vending machine that sells actually important items like guns and drills, I'm not sure about the price to select, so I just went for "lower than the astronomical mining equipment costs, while remembering 300 $ starting money and the loadse dosh you get from basically bothering to mine"
- Refilling the vending machine is as simple as ordering a vending pack bundle (two mining packs) and carrying it over to your vending machines if you really managed to empty them
- You can also make new vending machines this way, which used to be completely impossible with mining equipment outside of admin spawn

Again, THIS IS NOT A BALANCE CHANGE. The whole point of this PR is to change the premium snowflake that is the Mining Equipment vendor and change it into an actual vending machine with the exact same functionality

**Comment by Shadowmech88** [comment]
It wouldn't be a balance change if literally all you were doing is making it behave like a vending machine. That would be fine.

Instead you're ALSO changing what can be bought, how much the things cost, and who can use the machine. Those are all balance changes.
If you really want to make the equipment vendor into a vending machine, I think the PR would go through a lot more smoothly if you didn't try to include these balance changes in a PR which you yourself say is not a balance PR.

**Comment by dylanstrategie** [comment]
All prices are back to default. Are we where we want to be, or will I have to emulate every last fucking pixel of this thing until it would be impossible to tell it was changed without access to the code ?

**Comment by d3athrow** [comment]
@dylanstrategie the issue is your changes always end up fucking with balance because you touch things that don't need to be touched.

**Comment by d3athrow** [comment]
damn you tab

**Comment by Shadowmech88** [comment]
@dylanstrategie Can it be used without mining access as well?

**Comment by dylanstrategie** [comment]
Since it is a vending machine, you can snap the ID wire to access it as any crewman. And since Cargo is nearby, insulated gloves and multitools shouldn't be hard to come by unless you are pillaging Cargo

**Comment by Shadowmech88** [comment]
It's still a balance change, though. I'm not saying it should or shouldn't be done, but that'll make it a balance PR instead of a refactor PR.

**Comment by Zth--** [comment]
Guys, guys, calm down. Soaps can be made at hydroponics (bio machine).

**Comment by dylanstrategie** [comment]
Jesus fucking christ

>"Map conflicts don't happen if you modify different lines"

As if

**Pull request closed without merging**